apps/sms/js/threads.js

```diff
@@ -86,12 +89,22 @@
 
   var Threads = exports.Threads = {
     registerMessage: function(message) {
-      var thread = Thread.create(message);
-      var threadId = message.threadId;
+      var threadId = +message.threadId;
+      var messageId = +message.id;
+
       if (!this.has(threadId)) {
-        this.set(threadId, thread);
+        this.set(threadId, Thread.create(message));
+      }
+      this.get(threadId).messages.set(messageId, message);
+      messageMap.set(messageId, threadId);
+    },
+    unregisterMessage: function(id = +id) {
```

Comment:
I'm sorry, this doesn't work... the right part is used only when the argument is not provided when you call the function...